akka-http-core/src/main/scala/akka/http/scaladsl/Http2Shadow.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import akka.actor.{ ActorSystem, ExtendedActorSystem, ExtensionId }
+import akka.annotation.InternalApi
+import akka.event.LoggingAdapter
+import akka.http.scaladsl.Http.ServerBinding
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
+import akka.http.scaladsl.settings.ServerSettings
+import akka.stream.{ ActorMaterializerHelper, Materializer }
+
+import scala.concurrent.Future
+
+/** INTERNAL API: Uses reflection to reach for Http2 support if available or fails with an exception */
+@InternalApi
+private[akka] object Http2Shadow {
+
+  type ShadowHttp2 = AnyRef
```

Comment:
In HTTP2 land we might be able to add a compilation test that does nothing more than

`Http2: Http2Shadow.ShadowHttp2`

so that we will catch if the signatures would get out of sync.